allennlp/data/dataset_readers/semantic_role_labelling.py

```diff
@@ -0,0 +1,228 @@
+    8 Predicate Frameset ID: int
+        The PropBank frameset ID of the predicate in Column 7.
+    9 Word sense: float
+        This is the word sense of the word in Column 3.
+    10 Speaker/Author: str
+        This is the speaker or author name where available. Mostly in Broadcast Conversation
+        and Web Log data. When not available the rows are marked with an "-".
+    11 Named Entities: str
+        These columns identifies the spans representing various named entities. For documents
+        which do not have named entity annotation, each line is represented with an "*".
+    12+ Predicate Arguments: str
+        There is one column each of predicate argument structure information for the predicate
+        mentioned in Column 7. If there are no predicates tagged in a sentence this is a
+        single column with all rows marked with an "*".
+    -1 Co-reference: str
+        Co-reference chain information encoded in a parenthesis structure. For documents that do
+         not have co-reference annotations, each line is represented with a "-".
+
+    Parameters
+    ----------
+    ontonotes_filename : ``str``
+    token_indexers : ``Dict[str, TokenIndexer]``, optional (default=``{"tokens": SingleIdTokenIndexer()}``)
+        We similarly use this for both the premise and the hypothesis.  See :class:`TokenIndexer`.
+
+    Return
+    ------
+    A ``Dataset`` of ``Instances`` for Semantic Role Labelling.
+
+    """
+    def __init__(self,
+                 ontonotes_filename: str,
+                 token_indexers: Dict[str, TokenIndexer] = None) -> None:
+        self._ontonotes_filename = ontonotes_filename
+        self._token_indexers = token_indexers or {"tokens": SingleIdTokenIndexer()}
+
+    def process_sentence(self,
+                         sentence: List[str],
+                         verbal_predicates: List[int],
+                         predicate_argument_labels: List[List[str]]) -> List[Instance]:
+        """
+        Parameters
+        ---------
+        sentence : List[str], required.
+            The tokenised sentence.
+        verbal_predicates : List[int], required.
+            The indexes of the verbal predicates in the
+            sentence which have an associated annotation.
+        predicate_argument_labels : List[List[str]], required.
+            A list of predicate argument labels, one for each verbal_predicate. The
+            internal lists are of length: len(sentence).
+
+        Return
+        ------
+        A list of Instances.
+
+        """
+        sentence_field = TextField(sentence, self._token_indexers)
+        if not verbal_predicates:
+            # Sentence contains no predicates.
+            tags = TagField(["O" for _ in sentence], sentence_field)
+            verb_indicator = IndexField(None, sentence_field)
+            instance = Instance(fields={"tokens": sentence_field, "verb_indicator": verb_indicator, "tags": tags})
```

Comment:
Should we just leave out the `verb_indicator` key, instead of having an `IndexField` with a `None` value?  I guess if we go that route, a lot of code in the instance processing pipeline has to change to account for different keys...  If you say `"verb_indicator": None` in the dictionary, we still need special casing in parts of the instance pipeline...  So, yeah, this is simpler.  Not sure that I like it, though...